SonarQube.TeamBuild.PreProcessor/ArgumentProcessor.cs

```diff
@@ -15,91 +15,25 @@
-    /// Process and validates the command line arguments and reports any errors
+    /// Process and validates the pre-processor command line arguments and reports any errors
     /// </summary>
-    /// <remarks>The command line parsing makes a number of simplying assumptions:
-    /// * order is unimportant
-    /// * all arguments have a recognisable prefix e.g. /key= 
-    /// * prefixes are case-insensitive
-    /// * unrecognised arguments are treated as errors
-    /// * the command line arguments are those supplied in Main(args) i.e. they have been converted
-    ///   from a string to an array by the runtime. This means that quoted arguments will already have
-    ///   been partially processed so a command line of:
-    ///        myApp.exe "quoted arg" /k="ab cd" ""
-    ///   will be supplied as three args, [quoted arg] , [/k=ab cd] and String.Empty</remarks>
     public static class ArgumentProcessor // was internal
     {
-        /// <summary>
-        /// Ids for supported arguments
-        /// </summary>
-        private enum KeywordIds
-        {
-            ProjectKey = 1,
-            ProjectName,
-            ProjectVersion,
-            RunnerPropertiesPath,
-            DynamicSetting
-        }
-
         #region Argument definitions
 
-        private static IList<ArgumentDescriptor> Descriptors;
-
         /// <summary>
-        /// Describes a single valid argument type - id, prefixes, multiplicity etc
+        /// Ids for supported arguments
         /// </summary>
-        private class ArgumentDescriptor
+        private static class KeywordIds
         {
-            /// <summary>
-            /// The unique (internal) identifier for the argument
-            /// </summary>
-            public KeywordIds Id { get; set; }
-
-            /// <summary>
-            /// Any prefixes supported for the argument. This should include all of the characters that
-            /// are not to be treated as part of the value e.g. /key=
-            /// </summary>
-            public string[] Prefixes { get; set; }
-
-            /// <summary>
-            /// Whether the argument is mandatory or not
-            /// </summary>
-            public bool Required { get; set; }
-
-            /// <summary>
-            /// A short description of the argument that will be displayed to the user
-            /// e.g. /key= [SonarQube project key]
-            /// </summary>
-            public string Description { get; set; }
-
-            /// <summary>
-            /// True if the argument can be specified multiple times,
-            /// false if it can be specified at most once
-            /// </summary>
-            public bool AllowMultiple { get; set; }
+            public const string ProjectKey = "1";
+            public const string ProjectName = "2";
+            public const string ProjectVersion = "3";
+            public const string RunnerPropertiesPath = "4";
+            public const string DynamicSetting = "5";
```

Comment:
I changed the descriptor id from an enum to a string to break the dependency between the CommandLineParser and the pre-processor. I've changed the comparisons in the CommandLineParser to use a string comparer.